tracer/test/Datadog.Trace.Tests/Agent/ApiTests.cs

```diff
@@ -40,13 +40,62 @@ public async Task SendTraceAsync_200OK_AllGood()
             factoryMock.Setup(x => x.GetEndpoint(It.Is<string>(s => s == TracesPath))).Returns(new Uri("http://localhost/traces"));
             factoryMock.Setup(x => x.GetEndpoint(It.Is<string>(s => s == StatsPath))).Returns(new Uri("http://localhost/stats"));
 
-            var api = new Api(apiRequestFactory: factoryMock.Object, statsd: TestStatsdManager.NoOp, updateSampleRates: null, partialFlushEnabled: false, healthMetricsEnabled: false);
+            var api = new Api(factoryMock.Object, TestStatsdManager.NoOp, Mock.Of<IContainerMetadata>(), updateSampleRates: null, partialFlushEnabled: false, healthMetricsEnabled: false);
 
             await api.SendTracesAsync(new ArraySegment<byte>(new byte[64]), 1, false, 0, 0, false);
 
             requestMock.Verify(x => x.PostAsync(It.IsAny<ArraySegment<byte>>(), MimeTypes.MsgPack), Times.Once());
         }
 
+        [Theory]
+        [CombinatorialData]
+        public async Task SendWithContainerMetadata(
+            [CombinatorialValues(null, "my-container-id")]
+            string containerId,
+            [CombinatorialValues(null, "my-entity-id")]
+            string entityId)
+        {
+            var responseMock = new Mock<IApiResponse>();
```

Comment:
Just FYI, we have `TestApiRequest` and `TestApiResponse` which are much easier to work with than Moq IMO. The Moq in here predates those. No need to have to change, just an FYI